ogr/ogr_schema_override.h

```diff
@@ -0,0 +1,382 @@
+        {
+            osFieldsSchemaOverride = osJSON;
+        }
+
+        CPLJSONDocument oSchemaDoc;
+        if (oSchemaDoc.LoadMemory(osFieldsSchemaOverride))
+        {
+            const CPLJSONObject oRoot = oSchemaDoc.GetRoot();
+            if (oRoot.IsValid())
+            {
+                const auto aoLayers = oRoot.GetArray("layers");
+                // Loop through layer names and get the field details for each field.
+                for (const auto &oLayer : aoLayers)
+                {
+                    if (oLayer.IsValid())
+                    {
+                        const auto oLayerFields = oLayer.GetArray("fields");
+                        // Parse fields
+                        const auto osLayerName = oLayer.GetString("name");
+                        const auto osSchemaType =
+                            oLayer.GetString("schemaType");
+                        // Default schemaType is "Patch"
+                        const auto bSchemaFullOverride =
+                            CPLString(osSchemaType).tolower() == "full";
+                        OGRLayerSchemaOverride oLayerOverride;
+                        oLayerOverride.SetLayerName(osLayerName);
+                        oLayerOverride.SetFullOverride(bSchemaFullOverride);
+
+                        if (oLayerFields.Size() > 0 && !osLayerName.empty())
+                        {
+                            for (const auto &oField : oLayerFields)
+                            {
+                                const auto osFieldName =
+                                    oField.GetString("name");
+                                if (osFieldName.empty())
+                                {
+                                    CPLError(CE_Warning, CPLE_AppDefined,
+                                             "Field name is missing");
+                                    return false;
+                                }
+                                OGRFieldDefnOverride oFieldOverride;
+
+                                const auto oType =
+                                    CPLString(oField.GetString("type"))
+                                        .tolower();
+                                const auto oSubType =
+                                    CPLString(oField.GetString("subType"))
+                                        .tolower();
+                                const auto osNewName =
+                                    CPLString(oField.GetString("newName"))
+                                        .tolower();
+                                const auto nWidth = oField.GetInteger(
+                                    "width",
+                                    std::numeric_limits<int>::quiet_NaN());
+                                const auto nPrecision = oField.GetInteger(
+                                    "precision",
+                                    std::numeric_limits<int>::quiet_NaN());
```

Comment:
I had no idea that ``std::numeric_limits<int>::quiet_NaN()`` was valid... That looks super odd. Perhaps just 0 for clarify ?

```suggestion
                                    0);
                                const auto nPrecision = oField.GetInteger(
                                    "precision",
                                    0);
```

Reply:
done